inlet/flow/input/udp/root.go

```diff
@@ -185,6 +185,12 @@ func (in *Input) Start() (<-chan []*decoder.FlowMessage, error) {
 					Payload:      payload[:n],
 					Source:       source.IP,
 				})
+				// if the agent Id SrcAddrOverwrite is set, we actually overwrite the exporter Address
+				if in.config.AgentIDSrcAddrOverwrite {
+					for _, f := range flows {
+						f.ExporterAddress = source.IP
+					}
+				}
```

Comment:
This should be moved one level up, in the `inlet/flow` component. The `Decode()` methods also receive the source IP. This way, this would also work for PCAP files or for a future input method capturing from an interface.

Reply:
If i'm doing it in https://github.com/akvorado/akvorado/blob/28166569d26580ca8305a6b827aedb22a665d883/inlet/flow/root.go#L143 i don't have the respective input configuration any more. I'd keep the setting on a "per input configuration" however, to allow different settings for e.g. sflow/netflow/etc. - how do we solve this? We could try add the InputConfiguration to `type Input interface {`?

Reply:
I think you can do it here: https://github.com/akvorado/akvorado/blob/28166569d26580ca8305a6b827aedb22a665d883/inlet/flow/decoder.go#L27

Reply:
When doing it on that level, i'm having problems with the configuration at the moment:
For my understanding, this should go into 
```
// InputConfiguration represents the configuration for an input.
type InputConfiguration struct {
	// Decoder is the decoder to associate to the input.
	Decoder string
	// Config is the actual configuration of the input.
	Config input.Configuration
	// UseSrcAddrForExporterAddr replaces the exporter address by the UDP flow source.
	// default is false, meaning that the agent id inside the flowmessage is used
	UseSrcAddrForExporterAddr bool
}
``` 
to be generic for all input methods. I can access it in the decoder from there, however, it is not getting parsed correctly from the input configuration. How do I solve this?

Reply:
Keep it in UDP for now, I'll investigate this later (the solution is to also modify `ConfigurationUnmarshallerHook()` and `MarshalYAML()` functions, but I would like to do something more generic to avoid having to update these functions.

Reply:
Rolled back the change and tested it with the current setup, should work as it is now.